6_nodejs/week3/app.js

```diff
@@ -0,0 +1,53 @@
+import knexInstance from 'knex'
+const knex= knexInstance({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "my-secret-password",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+import 'dotenv/config'
+import express from 'express'
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knex("contacts").select();
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+    if (orderBy === "first_name"){
```

Comment:
the input query parameter will be first_name%20ASC, meaning sort by first name ascending. so orderby check here would be 
`if (orderBy === "first_name ASC"){ //... }`

the same for orderBy last name.